### PR TITLE
Instantiate Queue via Chat Command

### LINKED ISSUE #8: Instantiate Queue via Chat Command
As an instructor, I want to instantiate a queue via chat command so that I can keep track of students with questions during my office hours.

### PR DESCRIPTION
**Pull request by richardcoucoules**
Fixes #8 - Instantiate Queue via Chat Command

This PR introduces a new Queue class found in `bot/utilities/Queue.ts`. In addition, a new data member, `activeQueue` was added to the `TeamsBot` class and the corresponding command `start office hour` was added to the manifest and message handler to instantiate a new Queue.

Currently, a new queue is instantiated with random values for its `id`, `ownerId`, and `channelId` properties. The `startTime` property value is set to the time of instantiation, the `status` property value is set to Active, and the `entries` property value is set to an empty array.

Edit: commit c470036 changes the random behavior: `id` is set to -1, `ownerId` is set to the ID of the user who issued the `start office hour` command, and `channelId` is set to the message handler's `context.activity` object's `channelId`, which should be the ID of the channel in which the message was received.

### CODE REVIEW

File: templates/appPackage/manifest.local.template.json [hunk 1]
```diff
@@ -40,6 +40,10 @@
             {
               "title": "learn",
               "description": "Learn about Adaptive Card and Bot Command"
+            },
+            {
+              "title": "start office hours",
```
Comment (rorochaudhary) [inline templates/appPackage/manifest.local.template.json]: I started running the app locally and was not able to get initialize office hours using the command. I think what may be happening here is that the switch case in teamsBot.ts is looking for input of "start office hour", while this remote template gives a menu option as "start office hour**s**". Manually typing in the command "start office hour" does initialize the queue and the bot responds. Fixing this should allow for the command in the menu to work!

**Comment by rorochaudhary** [comment]
Good call on the randomized values as a stub-in for what will come directly from Teams. Was able to run it locally, and while I think I may have found the issue for why the command menu does not work, that is a relatively minor fix compared to the core functionality that you got going here - meets the spec as we ironed out in the Project Plan. Good stuff!

**Comment by richardcoucoules** [comment]
After playing around with the bot, I was able to get the new command to show up by installing the new app package (`build/appPackage/appPackage.local.zip`) manually into my local Teams environment using the workflow shown in the picture below. It seems that launching the bot through the Teams debugger builds the app package correctly, but skips some essential step that uploads the updated manifest file.

image

File: templates/appPackage/manifest.local.template.json (hunk 1 above)
Comment (richardcoucoules) [reply templates/appPackage/manifest.local.template.json]: Thanks for the heads up! I fixed this in commit e432cc4 but it didn't do the trick for the new command. I found a way to force the update, which I've written out in https://github.com/beaver-hours/BeaverHours/pull/12#issuecomment-1015000853.

**Comment by rorochaudhary** [comment]
Pulling in the changes and running locally from the Teams debugger works on my end! The changes look good, thanks for adding in the channel and owner ids as well. Approved!

image

**Review by rorochaudhary — APPROVED** [review]
This approval comment is copied from the conversation on this PR as I misplaced it:
------
Pulling in the changes and running locally from the Teams debugger works on my end! The changes look good, thanks for adding in the channel and owner ids as well. Approved!

**Pull request merged**